File: 0057-insert-interval/0057-insert-interval.cpp
class Solution {
private:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> ans;
        
        for(int i=0;i<intervals.size();i++){
                if(ans.empty()) ans.push_back(intervals[i]);
                else if(ans.back()[1]>=intervals[i][0]){
                    if(ans.back()[1]>=intervals[i][1]) continue;
                    else ans.back()[1]=intervals[i][1];
                }
                else ans.push_back(intervals[i]);
            }   
        return ans;
    }
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        vector<vector<int>> newintervals;
        if(intervals.size()==0) return {newInterval};
        //insert in sorted place
        for(int i=0;i<intervals.size();i++){
            if(intervals[i][0]>newInterval[0]){
                newintervals.push_back(newInterval);
            }
            newintervals.push_back(intervals[i]);
        }
            if(newintervals.size()!=intervals.size()+1) newintervals.push_back(newInterval);
            return merge(newintervals);
        }
};

File: 0057-insert-interval/0057-insert-interval.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        res = []
        n = len(intervals)

        for i in range(n):

            if newInterval[0] > intervals[i][1]:
                res.append(intervals[i])

            elif newInterval[1] < intervals[i][0]:
                res.append(newInterval)
                return res + intervals[i:]
            
            else:
                newInterval = [
                    min(newInterval[0],intervals[i][0]),
                    max(newInterval[1],intervals[i][1])
                    ]


            
        res.append(newInterval)

        return res

